### PR TITLE
RAC5071: failed to copy artifact of PACKER_CACHE_BUILD

### PR DESCRIPTION
**Pull request by PengTian0**
The PR fix the issue by:

- split job PACKER_CACHE_BUILD into 2 jobs: OVA_CACHE_BUILD and VAGRANT_CACHE_BUILD

- MasterCI builds ova only when job OVA_CACHE_BUILD  is not running .

- MasterCI builds vagrant only when job VAGRANT_CACHE_BUILD  is not running.

### CODE REVIEW

**Comment by PengTian0** [comment]
@panpan0000 @changev @anhou @iceiilin

**Comment by PengTian0** [comment]
Jobs:
test for MasterCI: http://rackhdci.lss.emc.com/job/RAC5071/
OVA_CACHE_BUILD: http://rackhdci.lss.emc.com/job/OVA_CACHE_BUILD/
VAGRANT_CACHE_BUILD: http://rackhdci.lss.emc.com/job/VAGRANT_CACHE_BUILD/

File: jobs/build_ova/build_ova.groovy [hunk 1]
```diff
@@ -6,53 +6,55 @@ node{
...
-                    checkout scm
-                }
-                def url = "https://github.com/RackHD/RackHD.git"
-                def branch = "${env.RACKHD_COMMIT}"
-                def targetDir = "build"
-                shareMethod.checkout(url, branch, targetDir)
-                step ([$class: 'CopyArtifact',
-                projectName: 'PACKER_CACHE_BUILD',
-                target: 'cache_image']);
+lock("ova_build"){
```
Comment (panpan0000) [inline jobs/build_ova/build_ova.groovy:9]: I think the lock can only apply to copy-artifact , aka, the build_ova can live outside the lock scope.

File: jobs/build_vagrant/cleanup_vbox.sh [hunk 2]
```diff
@@ -9,7 +9,8 @@ virtualBoxDestroyAll() {
         vboxmanage controlvm ${uuid} poweroff
         echo "deleting vm ${uuid}"
         vboxmanage unregistervm ${uuid}
-      done
+    done
+    pkill packer
```
Comment (panpan0000) [inline jobs/build_vagrant/cleanup_vbox.sh:13]: Thank you

File: jobs/build_ova/build_ova.groovy (hunk 1 above)
Comment (panpan0000) [reply jobs/build_ova/build_ova.groovy:9]: talk with @PengTian0 
there's re 2 options of the locking . and option 1 was what I intended. but she chose option 2

per @PengTian0 , 
option 1:
Pros : smaller lock , only lock when necessary. reasonable code logic.
Cons:  when a ova-build job has locked slave, but waiting resource lock(cache build ongoing). But the slave will be idle. and there's 'ova-post-test' requires same slaves will wait. 
option 2:
Pros : only lock slave if cache is ready( resource lock acquired)
Cons : seems bigger lock scope 


**so Tian think make slave used more effective is of more priority**



option 1:
```
lock_slave()
lock_resource()  // the lock between OVA_CACHE_BUILD & OVA_BUILD
copy_from_cache_archiving()
unlock_resource()
do_ova_build()
unlock_slave()
```

option 2:
```
lock_resource()  // the lock between OVA_CACHE_BUILD & OVA_BUILD
lock_slave()
copy_from_cache_archiving()
do_ova_build()
unlock_slave()
unlock_resource()
```

**Review by panpan0000 — APPROVED** [review]

**Review by anhou — APPROVED** [review]

**Pull request merged**